### PR TITLE
Exceptions for audits should only get wrapped if audit mode is enabled

### PR DESCRIPTION
**Pull request by jdmundrawala**
cc @jtimberman @sersut @mcquin

### CODE REVIEW

**Comment by mcquin** [comment]
Wrapping chef client run errors helps with consistency in the format of our
error messaging. I think it could be confusing to have two separate errors
reported (the error, and the wrapped error) depending on whether or not
audit mode is enabled. And for users who care what kind of error is thrown
(if such users exist?), it would be annoying to have to know that you have
to check for the error or a RunFailedWrappingError containing the error.

**Comment by jdmundrawala** [comment]
@mcquin I think that this would need a major version bump:
https://github.com/chef/chef/issues/3054
https://github.com/chef/chef/issues/3078

**Comment by mcquin** [comment]
That doesn't address my concern of breaking consistency in our error
messaging.

What I think we should be doing, is proper exit coding based off of the
contained errors.

**Comment by jdmundrawala** [comment]
@mcquin that's a fair point, however we should wait until chef 13 to make this change. We counted it as a regression in 12.2, and we know it will break people by wrapping the exception

**Comment by mcquin** [comment]
OK. That this was resolving the regression wasn't clear.

Where can we make a note that in Chef 13 we should unify run errors and fix
exit codes?

File: spec/unit/client_spec.rb [hunk 1]
```diff
@@ -512,11 +513,26 @@ def stub_for_sync_cookbooks
...
+      it "should run exception handlers on early fail" do
+        expect(subject).to receive(:run_failed)
+        expect { subject.run }.to raise_error(Chef::Exceptions::RunFailedWrappingError) do |error|
+          expect(error.wrapped_errors.size).to eq 1
+          expect(error.wrapped_errors).to include(NoMethodError)
+        end
+      end
+    end
+
+    context 'when audit mode is enabled' do
```
Comment (tyler-ball) [inline spec/unit/client_spec.rb]: `'when audit mode is disabled'`

File: spec/unit/client_spec.rb [hunk 2]
```diff
@@ -238,23 +238,24 @@ def stub_for_sync_cookbooks
     describe "when converge completes successfully" do
       include_context "a client run"
       include_context "converge completed"
-
-      describe "when audit phase errors" do
-        include_context "audit phase failed with error"
-        include_examples "a completed run with audit failure" do
-          let(:run_errors) { [audit_error] }
+      context 'when audit mode is not disabled' do
```
Comment (tyler-ball) [inline spec/unit/client_spec.rb]: (nit) double negatives are bad, mmm'kay.  `'when audit mode is enabled'`

**Comment by tyler-ball** [comment]
Some minor comments, but overall :+1: for fixing the regression.  We should be able to add a test with the `:chef_gte_13_only` flag and assert that even when audit mode is disabled we have the wrapped error.  We can totally change this behavior in Chef 13, but at least then we would have a failing test to remind us that we need to have a discussion.

File: spec/unit/client_spec.rb (hunk 2 above)
Comment (jdmundrawala) [inline spec/unit/client_spec.rb]: m'kay

**Comment by ranjib** [comment]
:+1:

**Pull request merged**